store/sqlstore/upgrade.go

```diff
@@ -743,5 +732,17 @@ func UpgradeDatabaseToVersion516(sqlStore SqlStore) {
 			sqlStore.GetMaster().Exec("ALTER TABLE Tokens MODIFY Extra text")
 		}
 		saveSchemaVersion(sqlStore, VERSION_5_16_0)
+
+		// Fix mismatches between the canonical and migrated schemas.
+		sqlStore.AlterColumnTypeIfExists("TeamMembers", "SchemeGuest", "tinyint(4)", "boolean")
+		sqlStore.AlterColumnTypeIfExists("ChannelMembers", "SchemeGuest", "tinyint(4)", "boolean")
```

Comment:
I appreciate this isn't the actual original PR making these changes, but this concerns me. Will it do a full rewrite of the `ChannelMembers` table? We're explicitly trying to avoid any expensive change to the schema of tables like this and `Posts` because of the potential for extended downtime on upgrade for customers.

Reply:
In the absence of certainty, I'd want to know how long it takes to run on a ChannelMembers table with a few million entries in it.

Reply:
At least with respect to MySQL 5.7 and as per https://dev.mysql.com/doc/refman/5.7/en/innodb-online-ddl-operations.html:
* Adding a column rebuilds the table in replace, but allows concurrent DML operations
* Changing the column data type rebuilds the table, but not in place, and disallows concurrent DML operations.

So it's almost certainly at least as expensive as when the column was added earlier this year, but likely adds its own overhead. I'm assuming Postgres to be no worse (if not just better).

If we're not comfortable making this change, we'll need to revert from v5.16 (at least this column), and acknowledge this column's definition will be permanently different for different clients. It shouldn't have any semantic difference.

Reply:
As I understood it at the time (although now struggling to find the source of this info), the cost of adding a column was less *iff* the column was nullable (perhaps this was postgres related). So we got away with adding it, but we wouldn't get away with modifying it.

Given where we are in the release timeline, my preference would be to revert this and verify it explicitly for 5.18, assuming it is not a huge amount of work to revert?

Reply:
Sounds good -- I've reverted this one fix, and updated our `make test-db-migration` to compare MySQL with the column dropped.